File: string/0929_unique_email_addresses/0929_unique_email_addresses.py
from typing import List

class Solution:
    def numUniqueEmails(self, emails: List[str]) -> int:
        seen = set()
        for email in emails:
            local, domain = email.split('@')
            if '+' in local:
                local = local[:local.index('+')]
            seen.add(local.replace('.','') + '@' + domain)
        return len(seen)



emails = ["test.email+alex@leetcode.com","test.e.mail+bob.cathy@leetcode.com","testemail+david@lee.tcode.com"]
res = Solution().numUniqueEmails(emails)
print(res)

File: string/0929_unique_email_addresses/0929_unique_email_addresses.go
package main

import (
	"fmt"
	"strings"
)

func numUniqueEmails(emails []string) int {
	m := map[string]struct{}{}
	for _, email := range emails {
		i := strings.Index(email, "+")
		j := strings.LastIndex(email, "@")
		if i == -1 {
			i = j
		}
		tmp := strings.ReplaceAll(email[:i], ".", "") + email[j:]
		m[tmp] = struct{}{}
	}
	return len(m)
}

func main() {
	emails := []string{"test.email+alex@leetcode.com", "test.e.mail+bob.cathy@leetcode.com", "testemail+david@lee.tcode.com"}
	res := numUniqueEmails(emails)
	fmt.Println(res)
}
